### ISSUE TITLE
[Accessibility] Focus issues with Share dialog

### ISSUE THREAD

**Issue by PeterSaathoff-Harshfield** [ampproject/amp.dev#4498]
Environment:
Any OS/browser and screen reader (such as VoiceOver, NVDA, or ChromeVox)

Steps to reproduce:
1. Open https://amp-dev-staging.appspot.com/page-experience/
2. Enter a valid URL and press Analyze
3. Turn on a screen reader
4. Navigate with the screen reader to the Share button
5. Activate the Share button
6. Attempt to navigate with the screen reader to the dialog

In my testing, NVDA, iOS VoiceOver, macOS VoiceOver, and ChromeVox screen readers were all unable to navigate to the dialog.

7. Turn off the screen reader and use the keyboard (tab, shift-tab, enter) to navigate the dialog

When the dialog is open, pressing tab should not close the dialog. It should cycle keyboard focus through focusable controls in the dialog, until the close button or escape is pressed. All controls in the dialog should also be useable with the keyboard. Currently the close button, text field containing URL, and Copy to clipboard button are all not keyboard focusable.

[Creating An Accessible Modal Dialog](https://bitsofco.de/accessible-modal-dialog/)
